File: ddd/src/main/java/yun/blog/ddd/model/Address.java
package yun.blog.ddd.model;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import javax.validation.constraints.NotEmpty;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.util.Assert;

@Embeddable
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Address {

    @NotEmpty
    @Column(name = "address1", nullable = false)
    private String address1;

    @NotEmpty
    @Column(name = "address2", nullable = false)
    private String address2;

    @NotEmpty
    @Column(name = "zip", nullable = false)
    private String zip;

    @Builder
    public Address(String address1, String address2, String zip) {
        Assert.hasText(address1, "address1 must not be empty");
        Assert.hasText(address2, "address1 must not be empty");
        Assert.hasText(zip, "address1 must not be empty");

        this.address1 = address1;
        this.address2 = address2;
        this.zip = zip;
    }
}

File: yun/domain/src/main/kotlin/com/cheese/yun/domain/model/Address.kt
package com.cheese.yun.domain.model

import javax.persistence.Column
import javax.persistence.Embeddable

@Embeddable
data class Address(
    @Column(name = "zip_code", nullable = false)
    val zipCode: String,
    @Column(name = "city", nullable = false)
    val city: String,
    @Column(name = "detail_address", nullable = false)
    val detailAddress: String
)